### PR TITLE
fix: handle site geometry null case

### PR DESCRIPTION
**Pull request by sunilsabatp**
This PR addresses the case where a `project site geometry is null`, ensuring that the application handles this scenario gracefully without causing errors or unexpected behavior.

### CODE REVIEW

**Review by mariahosfeld — CHANGES_REQUESTED** [review]
For projects without sites I don't encounter any issues.
But when checking out a project with sites and treemapper data (/yucatan), I first see the vegetation layer and it takes a while (compared to the base) until the intended map with treemapper data shows up. Don't know if this is related, so please check and if needed amend.

**Review by mohitb35 — APPROVED** [review]
LGTM

**Comment by mohitb35** [comment]
> For projects without sites I don't encounter any issues. But when checking out a project with sites and treemapper data (/yucatan), I first see the vegetation layer and it takes a while (compared to the base) until the intended map with treemapper data shows up. Don't know if this is related, so please check and if needed amend.

I see this issue too.

**Comment by sunilsabatp** [comment]
> > For projects without sites I don't encounter any issues. But when checking out a project with sites and treemapper data (/yucatan), I first see the vegetation layer and it takes a while (compared to the base) until the intended map with treemapper data shows up. Don't know if this is related, so please check and if needed amend.
> 
> I see this issue too. May not show up on local, unless you disable `reactStrictMode`

I have resolve this issue in this PR [projects-redesign/facelift-feedback](https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/2274)

**Review by mariahosfeld — COMMENTED** [review]
Now the sat layer is not shown when selecting /yucatan, but still a quite long latency for TreeMapper data to show. Maybe it is because of a first load of the project, but we need to keep an eye out and think about how we could optimize it.

File: src/features/projectsV2/ProjectsContext.tsx [hunk 1]
```diff
@@ -271,7 +271,7 @@ export const ProjectsProvider: FC<ProjectsProviderProps> = ({
     projectSlug: string,
     siteIndex: number | null
   ) => {
-    if (!singleProject?.sites?.length) return;
+    if (!singleProject?.sites) return;
```
Comment (mohitb35) [inline src/features/projectsV2/ProjectsContext.tsx:274]: I'm not sure why this was changed, but the sites array could be empty. In that case, we should not be able to set a selectedSite or add the site query param in the url

**Review by mohitb35 — APPROVED** [review]
@sunilsabatp Minor feedback, can address that and merge

**Review by mohitb35 — APPROVED** [review]
LGTM

**Comment by mohitb35** [comment]
@sunilsabatp 
1. While visiting `/prd/trees-for-peace-an-ayudh-europe-campaign`, the site with the null geometry remains in the url.
2. While visiting `/prd/mothersforest?site=asdfkjasfjsdnfjksdn`, the site param remains in the url even though the site is not found.

**Review by mariahosfeld — COMMENTED** [review]
Functionality looks good to me.

**Review by mohitb35 — APPROVED** [review]
LGTM

**Pull request merged**